examples/app-basic/src/components/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, ChangeEvent } from 'react';
+import { Button, Legend, Field, Input, useStyles2 } from '@grafana/ui';
+import { PluginConfigPageProps, AppPluginMeta, PluginMeta, GrafanaTheme2 } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import { css } from '@emotion/css';
+import { SecretInput } from 'components/SecretInput';
+
+export type JsonData = {
+  apiUrl?: string;
+  isApiKeySet?: boolean;
+};
+
+type State = {
+  // The URL to reach our custom API.
+  apiUrl: string;
+  // Tells us if the API key secret is set.
+  // Set to `true` ONLY if it has already been set and haven't been changed.
+  // (We unfortunately need an auxiliray variable for this, as `secureJsonData` is never exposed to the browser after it is set)
+  isApiKeySet: boolean;
+  // An secret key for our custom API.
+  apiKey: string;
+};
+
+interface Props extends PluginConfigPageProps<AppPluginMeta<JsonData>> {}
+
+export const AppConfig = ({ plugin }: Props) => {
+  const s = useStyles2(getStyles);
+  const { enabled, pinned, jsonData } = plugin.meta;
+  const [state, setState] = useState<State>({
+    apiUrl: jsonData?.apiUrl || '',
+    apiKey: '',
+    isApiKeySet: Boolean(jsonData?.isApiKeySet),
+  });
+
+  const onResetApiKey = () =>
+    setState({
+      ...state,
+      apiKey: '',
+      isApiKeySet: false,
+    });
+
+  const onChangeApiKey = (event: ChangeEvent<HTMLInputElement>) => {
+    setState({
+      ...state,
+      apiKey: event.target.value.trim(),
+    });
+  };
+
+  const onChangeApiUrl = (event: ChangeEvent<HTMLInputElement>) => {
+    setState({
+      ...state,
+      apiUrl: event.target.value.trim(),
+    });
+  };
+
+  return (
+    <div className="gf-form-group">
```

Comment:
nit: we should try to stay away from the deprecated `gf-` classes. Especially in the examples where people might copy parts from it.

Reply:
Good point, will fix 👍

Reply:
Fixed in https://github.com/grafana/grafana-plugin-examples/pull/26/commits/f0187dc80138785826986fbb33b66574080daa30.